### ISSUE TITLE
`addBuildOption` results in `error: unable to find 'build_options'`

### ISSUE THREAD

**Issue by prime31** [ziglang/zig#5375]
When setting a build option in `build.zig` via `exe.addBuildOption(bool, "the_option", true)` the package isnt available from code in the `exe` it was added to.

Adding this to a file in the exe:
```const build_options = @import("build_options");```
Result in this error:
```error: unable to find 'build_options'```

The build command seems to properly contain the pkg definition:
```
zig build-exe /Users/mikedesaro/Zig-Aya/examples/thing.zig --library c++ --library c --library SDL2 --pkg-begin build_options SNIPPED/zig-cache/run_build_options.zig --pkg-end SNIPPED
```

**Comment by dbandstra** [comment]
Are you trying to import "build_options" in a zig file that's imported via a path set up using addPackagePath in build.zig?

**Comment by prime31** [comment]
Yes. I believe the actual issue is what you describe. addPackagePath calls seem to not work from other addPackagePath calls.

**Comment by floooh** [comment]
@prime31 I just stumbled over the same issue, as a workaround it seems to be possible to define a package dependency in the build.zig file using the code-generated xxx_buildoptions.zig file in zig-cache like this:

```zig
    const pkg_buildoptions = Pkg{
        .name = "build_options", 
        .path = switch (kc85_model) {
            .KC85_2 => "zig-cache/kc852_build_options.zig",
            .KC85_3 => "zig-cache/kc853_build_options.zig",
            .KC85_4 => "zig-cache/kc854_build_options.zig"
        },
    }
```

...and then make this package available to other non-root packages:

```zig
    const pkg_emu = Pkg{
        .name = "emu",
        .path = "src/emu/emu.zig", 
        .dependencies = &[_]Pkg{ pkg_buildoptions}
    }
```

I guess the next best option (apart from making the buildin_options package available to all project packages) would be to access the "magic" buildoptions Pkg object on the LibExeObjStep object in order to get rid of the zig-cache path hack.
